src/gen.ts

```diff
@@ -244,6 +271,28 @@ export function processCXType(type: CXType): TypeMetadata {
     canonical: type.getCanonicalType().getSpelling(),
     canonicalKind: type.getCanonicalType().getKindSpelling(),
     size: type.getSizeOf(),
+    elementType: type.getArrayElementType()
+      ? processCXType(type.getArrayElementType()!)
+      : undefined,
+    pointeeType: type.getPointeeType()
+      ? processCXType(type.getPointeeType()!)
+      : undefined,
+    arraySize: type.getArraySize() < 0 ? undefined : type.getArraySize(),
+    block: type.getCanonicalType().getKindSpelling() === "BlockPointer"
+      ? {
+        returnType: type.getPointeeType()?.getResultType()
+          ? processCXType(type.getPointeeType()?.getResultType()!)
+          : undefined,
+        parameters: Array.from(
+          {
+            length: type.getPointeeType()?.getNumberOfArgumentTypes() ?? 0,
+          },
+          (_, i) => {
+            return processCXType(type.getPointeeType()!.getArgumentType(i)!);
+          },
+        ),
+      }
+      : undefined,
   };
```

Comment:
GitHub doesn't let me add a diff for previously untouched lines, so just to say I think this looks a tiny bit cleaner by storing some of these variables in advance so that we can avoid using `?` or `!` on a few more things:

```ts
export function processCXType(type: CXType): TypeMetadata {
  const pointeeType = type.getPointeeType();
  const canonicalType = type.getCanonicalType();
  const arrayElementType = type.getArrayElementType();

  return {
    name: type.getSpelling(),
    kind: type.getKindSpelling(),
    canonical: canonicalType.getSpelling(),
    canonicalKind: canonicalType.getKindSpelling(),
    size: type.getSizeOf(),
    elementType: arrayElementType ? processCXType(arrayElementType) : undefined,
    pointeeType: pointeeType ? processCXType(pointeeType) : undefined,
    arraySize: type.getArraySize() < 0 ? undefined : type.getArraySize(),
    block:
      canonicalType.getKindSpelling() === "BlockPointer"
        ? {
            returnType: pointeeType?.getResultType()
              ? processCXType(pointeeType.getResultType()!)
              : undefined,
            parameters: Array.from(
              {
                length: pointeeType?.getNumberOfArgumentTypes() ?? 0,
              },
              (_, i) => {
                return processCXType(pointeeType!.getArgumentType(i)!);
              }
            ),
          }
        : undefined,
  };
}
```